module/VuFind/src/VuFind/Db/Service/SearchService.php

```diff
@@ -161,22 +158,37 @@ public function getSearchByIdAndOwner(
      */
     public function getSearches(?string $sessionId, UserEntityInterface|int|null $userOrId = null): array
     {
-        // If we don't get a session id or user id, don't return anything:
-        if (null === $sessionId && null === $userOrId) {
+
+        $userId = $userOrId instanceof UserEntityInterface ? $userOrId->getId() : $userOrId;
+
+        if (!$sessionId && !$userId) {
```

Comment:
The original code used explicit null checks, and your new code uses "truthiness" more broadly. I don't think it's likely to make a meaningful difference, but I'm curious if there's a reason for the change in strategy. (Usually we tend to move toward being more explicit rather than less when revising code).